docs/azure/sdk/authentication/create-token-credentials-from-configuration.md

```diff
@@ -0,0 +1,100 @@
+---
+title: Create token credentials from configuration
+description: This article describes how to create token credentials from configuration files
+ms.topic: how-to
+ms.custom: devx-track-dotnet, engagement-fy23
+ms.date: 1/16/2024
+---
+
+# Create Azure credential types using configuration files
+
+The `Microsoft.Extensions.Azure` library supports creating different <xref:Azure.Core.TokenCredential?displayProperty=fullName> types from values defined in _appsettings.json_ and other configuration files. This article describes the support for different `TokenCredential` types and how to configure the required values for each type.
+
+## Support for Azure credentials through configuration
+
+The `Microsoft.Extensions.Azure` library can automatically provide Azure service clients with a `TokenCredential` class by searching _appsettings.json_ or other configuration files for credential values using the default `IConfiguration` service for .NET. This approach allows developers to explicitly set credential values across different environments through configuration rather than through app code directly.
+
+The following credential types are supported via configuration:
+
+* <xref:Azure.Identity.ManagedIdentityCredential?displayProperty=fullName>
+* <xref:Azure.Identity.WorkloadIdentityCredential?displayProperty=fullName>
+* <xref:Azure.Identity.ClientSecretCredential?displayProperty=fullName>
+* <xref:Azure.Identity.ClientCertificateCredential?displayProperty=fullName>
+* <xref:Azure.Identity.DefaultAzureCredential?displayProperty=fullName>
+
+The configuration file values are only used when the service client doesn't explicitly set an authentication mechanism. For example, the following code uses `IConfiguration` to search for values in _appsettings.json_ at run time because the <xref:Azure.Storage.Blobs.BlobServiceClient?displayProperty=fullName> does not specify credentials during instantiation:
+
+```csharp
+// No TokenCredential or access key provided - configuration files will be searched
+var blobServiceClient = new BlobServiceClient("<storage-account-name>");
```

Comment:
This is not correct.  Configuration awareness is not directly built-into the Azure SDK clients.  The string-based constructor only accepts a full connection string.  

The configuration-driven credential is only used by the Azure Extensions when creating clients registered for DI.   It essentially calls the `builder.UseCredential` method to replace the default credential used for global auth or the local `builder.WithCredential` method to override the global credential for a specific client instance.

_([src](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/extensions/Microsoft.Extensions.Azure/src/AzureClientFactoryBuilder.cs#L35))_